diff --git a/dlls/iphlpapi/iphlpapi_main.c b/dlls/iphlpapi/iphlpapi_main.c
index b315490b1f2..df170bbe7d4 100644
--- a/dlls/iphlpapi/iphlpapi_main.c
+++ b/dlls/iphlpapi/iphlpapi_main.c
@@ -45,9 +45,18 @@
 WINE_DEFAULT_DEBUG_CHANNEL(iphlpapi);
 
 #define CHARS_IN_GUID 39
+#define ADAPTER_INFO_PATH "/data/data/com.termux/files/usr/tmp/adapterinfo"
 
 static const WCHAR *device_tcpip = L"\\DEVICE\\TCPIP_";
 
+struct adapter_info 
+{
+    char name[32];
+    char ip_address[16];
+    char netmask[16];
+    char gateway[16];
+};
+
 DWORD WINAPI AllocateAndGetIpAddrTableFromStack( MIB_IPADDRTABLE **table, BOOL sort, HANDLE heap, DWORD flags );
 
 static const NPI_MODULEID *ip_module_id( USHORT family )
@@ -221,6 +230,57 @@ DWORD WINAPI CreateProxyArpEntry(DWORD dwAddress, DWORD dwMask, DWORD dwIfIndex)
   return ERROR_NOT_SUPPORTED;
 }
 
+static struct adapter_info* read_adapters_info_from_file( int *adapter_info_size ) 
+{
+    FILE *file;
+    struct adapter_info* adapters_info = NULL;
+    *adapter_info_size = 0;
+    
+    file = fopen( ADAPTER_INFO_PATH, "r" );
+    if (file != NULL) 
+    {
+        int col, index, list_size = 0;
+        char* value;
+        const size_t line_size = 128;
+        char *line = malloc( line_size );
+        struct adapter_info *adapter_info;
+        
+        while (fgets( line, line_size, file ) != NULL)
+        {
+            index = list_size++;
+            adapters_info = realloc( adapters_info, list_size * sizeof(struct adapter_info) );
+            adapter_info = &adapters_info[index];
+
+            col = 0;
+            value = strtok( line, "," );
+            while (value) 
+            {
+                switch (col) 
+                {
+                case 0:
+                    strcpy( adapter_info->name, value );
+                    break;
+                case 1:
+                    strcpy( adapter_info->ip_address, value );
+                    break;
+                case 2:
+                    strcpy( adapter_info->netmask, value );
+                    break;
+                case 3:
+                    strcpy( adapter_info->gateway, value );
+                    break;                    
+                }
+                value = strtok( NULL, "," );
+                col++;
+            }     
+        }
+        free( line );        
+        fclose( file );
+        *adapter_info_size = list_size;
+    }
+    return adapters_info;
+}
+
 static char *debugstr_ipv6(const struct sockaddr_in6 *sin, char *buf)
 {
     const IN6_ADDR *addr = &sin->sin6_addr;
@@ -1329,15 +1389,96 @@ err:
 ULONG WINAPI DECLSPEC_HOTPATCH GetAdaptersAddresses( ULONG family, ULONG flags, void *reserved,
                                                      IP_ADAPTER_ADDRESSES *aa, ULONG *size )
 {
-    IP_ADAPTER_ADDRESSES *info, **sorted;
-    DWORD err, needed, count;
-
+    int adapter_info_size;
+    struct adapter_info *adapters_info; 
+ 
     TRACE( "(%ld, %08lx, %p, %p, %p)\n", family, flags, reserved, aa, size );
 
-    if (!size) return ERROR_INVALID_PARAMETER;
+    if (!size) return ERROR_INVALID_PARAMETER;  
 
-    err = adapters_addresses_alloc( family, flags, &info, &count );
-    if (err) return err;
+    adapters_info = read_adapters_info_from_file( &adapter_info_size );
+    if (adapters_info != NULL && adapter_info_size > 0) 
+    {
+        DWORD i, needed;
+        const int dns_suffix_len = 7;
+        const int adapter_name_len = 39;
+        const int friendly_name_len = 64;
+        const int aa_len = sizeof(*aa) + dns_suffix_len + adapter_name_len + friendly_name_len;
+        
+        needed = adapter_info_size * (aa_len + sizeof(IP_ADAPTER_UNICAST_ADDRESS) + sizeof(SOCKADDR_IN) + sizeof(IP_ADAPTER_GATEWAY_ADDRESS) + sizeof(SOCKADDR_IN));
+        if (!aa || *size < needed)
+        {
+            *size = needed;
+            return ERROR_BUFFER_OVERFLOW;
+        }
+        
+        for (i = 0; i < adapter_info_size; i++) 
+        {
+            WCHAR *friendly_name = HeapAlloc( GetProcessHeap(), HEAP_ZERO_MEMORY,  friendly_name_len );
+            
+            TRACE( "file adapter %s %s %s %s\n", adapters_info[i].name, adapters_info[i].ip_address, adapters_info[i].netmask, adapters_info[i].gateway );
+            
+            MultiByteToWideChar( CP_ACP, 0, adapters_info[i].name, -1, friendly_name, friendly_name_len );
+            
+            aa->Next = aa + 1;
+            aa->Length = aa_len;
+            aa->IfIndex = i;
+            aa->DnsSuffix = (WCHAR *)L"box";
+            aa->AdapterName = (CHAR *)"{00000000-0000-0000-0000-000000000000}";
+            
+            aa->Description = friendly_name;
+            aa->FriendlyName = friendly_name;
+            aa->PhysicalAddressLength = 6;
+            memset(&aa->PhysicalAddress, 0, 6);
+            aa->Mtu = 1;
+            aa->IfType = IF_TYPE_IEEE80211;
+            aa->OperStatus = 1;
+            aa->TransmitLinkSpeed = 144400000;
+            aa->ReceiveLinkSpeed = 144400000;
+            aa->ConnectionType = NET_IF_CONNECTION_DEDICATED;
+
+            if (!(flags & GAA_FLAG_SKIP_UNICAST)) 
+            {
+                IP_ADAPTER_UNICAST_ADDRESS *addr;               
+                
+                addr = HeapAlloc( GetProcessHeap(), HEAP_ZERO_MEMORY,  sizeof(*addr) + sizeof(SOCKADDR_IN) );
+                addr->Next = NULL;
+                addr->Length = sizeof(*addr);
+                addr->Address.lpSockaddr = (SOCKADDR *)(addr + 1);
+                addr->Address.iSockaddrLength = sizeof(SOCKADDR_IN);
+                addr->Address.lpSockaddr->sa_family = AF_INET;
+                ((SOCKADDR_IN *)addr->Address.lpSockaddr)->sin_addr.s_addr = inet_addr( adapters_info[i].ip_address );
+                aa->FirstUnicastAddress = addr;
+            }
+            
+            if (flags & GAA_FLAG_INCLUDE_GATEWAYS) 
+            {
+                IP_ADAPTER_GATEWAY_ADDRESS *gw;
+                
+                gw = HeapAlloc( GetProcessHeap(), HEAP_ZERO_MEMORY,  sizeof(*gw) + sizeof(SOCKADDR_IN) );
+                gw->Next = NULL;
+                gw->Length = sizeof(*gw);
+                gw->Address.lpSockaddr = (SOCKADDR *)(gw + 1);
+                gw->Address.iSockaddrLength = sizeof(SOCKADDR_IN);
+                gw->Address.lpSockaddr->sa_family = AF_INET;
+                ((SOCKADDR_IN *)gw->Address.lpSockaddr)->sin_addr.s_addr = inet_addr( adapters_info[i].gateway );
+                aa->FirstGatewayAddress = gw;
+            }
+            
+            aa++;
+        }
+        
+        free( adapters_info );
+        aa[-1].Next = NULL;
+        return 0;
+    }
+    else 
+    {
+        IP_ADAPTER_ADDRESSES *info, **sorted;
+        DWORD err, needed, count;
+    
+        err = adapters_addresses_alloc( family, flags, &info, &count );
+        if (err) return err;
 
     needed = adapters_addresses_size( info );
     if (!aa || *size < needed)
@@ -1355,8 +1496,9 @@ ULONG WINAPI DECLSPEC_HOTPATCH GetAdaptersAddresses( ULONG family, ULONG flags,
         }
     }
 
-    adapters_addresses_free( info );
-    return err;
+        adapters_addresses_free( info );
+        return err;
+    }
 }
 
 /******************************************************************
