--- a/dlls/ntdll/unix/virtual.c	2026-10-03 21:52:29.811315917 +0000
+++ b/dlls/ntdll/unix/virtual.c	2026-10-03 21:52:29.815315917 +0000
@@ -163,9 +163,9 @@
 static void *address_space_start = (void *)0x10000;
 #endif
 #ifdef _WIN64
-static void *address_space_limit = (void *)0x7fffffff0000;  /* top of the total available address space */
-static void *user_space_limit    = (void *)0x7fffffff0000;  /* top of the user address space */
-static void *working_set_limit   = (void *)0x7fffffff0000;  /* top of the current working set */
+static void *address_space_limit = (void *)0x7fffff0000;  /* top of the total available address space */
+static void *user_space_limit    = (void *)0x7fffff0000;  /* top of the user address space */
+static void *working_set_limit   = (void *)0x7fffff0000;  /* top of the current working set */
 #else
 static void *address_space_limit = (void *)0xc0000000;
 static void *user_space_limit    = (void *)0x7fff0000;
@@ -552,7 +552,7 @@
     /* if we don't have a preloader, try to reserve the space now */
     reserve_area( (void *)0x000000010000, (void *)0x000068000000 );
     reserve_area( (void *)0x00007f000000, (void *)0x00007fff0000 );
-    reserve_area( (void *)0x7ffffe000000, (void *)0x7fffffff0000 );
+    reserve_area( (void *)0x7ffe000000, (void *)0x7fffff0000 );
 
 #endif
 }
--- a/loader/preloader.c	2026-10-03 21:52:29.819315917 +0000
+++ b/loader/preloader.c	2026-10-03 21:52:29.819315917 +0000
@@ -115,7 +115,7 @@
     { (void *)0x000000010000, 0x00100000 },  /* DOS area */
     { (void *)0x000000110000, 0x67ef0000 },  /* low memory area */
     { (void *)0x00007f000000, 0x00ff0000 },  /* 32-bit top-down allocations + shared user data */
-    { (void *)0x7ffffe000000, 0x01ff0000 },  /* top-down allocations + virtual heap */
+    { (void *)0x7ffe000000, 0x01ff0000 },  /* top-down allocations + virtual heap */
 #endif
     { 0, 0 },                            /* PE exe range set with WINEPRELOADRESERVE */
     { 0, 0 }                             /* end of list */
